src/librustc_mir/build/matches/simplify.rs

```diff
@@ -56,11 +56,17 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                                  -> Result<(), MatchPair<'pat, 'tcx>> {
         let tcx = self.hir.tcx();
         match *match_pair.pattern.kind {
-            PatternKind::AscribeUserType { ref subpattern, ref user_ty, user_ty_span } => {
+            PatternKind::AscribeUserType {
+                ref subpattern,
+                variance,
+                ref user_ty,
+                user_ty_span
+            } => {
                 candidate.ascriptions.push(Ascription {
```

Comment:
Here we could add a comment like:

Apply the type ascription to the value at `match_pair.place`, which is the value being matched, taking the variance field into account.

Reply:
Added.